x-pack/legacy/plugins/ml/public/data_frame_analytics/common/analytics.ts

```diff
@@ -212,27 +222,67 @@ export function getValuesFromResponse(response: RegressionEvaluateResponse) {
 
   return { meanSquaredError, rSquared };
 }
+interface RegressionResultsSearchBoolQuery {
+  bool: Dictionary<any>;
+}
+interface RegressionResultsSearchTermQuery {
+  term: Dictionary<any>;
+}
+
+export type RegressionResultsSearchQuery =
+  | RegressionResultsSearchBoolQuery
+  | RegressionResultsSearchTermQuery
+  | SavedSearchQuery;
+
+export function getEvalQueryBody({
+  resultsField,
+  isTraining,
+  searchQuery,
+  ignoreDefaultQuery,
+}: {
+  resultsField: string;
+  isTraining: boolean;
+  searchQuery?: RegressionResultsSearchQuery;
+  ignoreDefaultQuery?: boolean;
+}) {
+  let query: RegressionResultsSearchQuery = {
+    term: { [`${resultsField}.is_training`]: { value: isTraining } },
+  };
+
+  if (searchQuery !== undefined && ignoreDefaultQuery === true) {
+    query = searchQuery;
+  } else if (isRegressionResultsSearchBoolQuery(searchQuery)) {
+    const searchQueryClone = cloneDeep(searchQuery);
+    searchQueryClone.bool.must.push(query);
+    query = searchQueryClone;
+  }
+  return query;
+}
 
 export const loadEvalData = async ({
   isTraining,
   index,
   dependentVariable,
   resultsField,
   predictionFieldName,
+  searchQuery,
+  ignoreDefaultQuery,
 }: {
   isTraining: boolean;
   index: string;
   dependentVariable: string;
   resultsField: string;
   predictionFieldName?: string;
+  searchQuery?: RegressionResultsSearchQuery;
+  ignoreDefaultQuery?: boolean;
 }) => {
   const results: LoadEvaluateResult = { success: false, eval: null, error: null };
   const defaultPredictionField = `${dependentVariable}_prediction`;
   const predictedField = `${resultsField}.${
     predictionFieldName ? predictionFieldName : defaultPredictionField
   }`;
 
-  const query = { term: { [`${resultsField}.is_training`]: { value: isTraining } } };
+  const query = getEvalQueryBody({ resultsField, isTraining, searchQuery, ignoreDefaultQuery });
```

Comment:
When the query matches just one document, I've seen that `r_squared` is returned by the evaluate endpoint as `-Infinity`. Can you check if this is correct, and if so, `getValuesFromResponse` needs to handle this case as currently it throws an error when calling `toPrecision` and the stats show up as `--` (the mean_squared_error is a value and could be displayed).

Reply:
This is caused because if there is only one doc variance is 0 - this was then being divided by 0 in the calculation to get rSquared. Instead of dividing by 0,  0 should just be returned. 

A fix for this is up already https://github.com/elastic/elasticsearch/pull/49439

We shouldn't need to do any additional checking in the UI.